topdown/tokens_test.go

```diff
@@ -117,19 +112,41 @@ OHoCIHmNX37JOqTcTzGn2u9+c8NlnvZ0uDvsd1BmKPaUmjmm
 			t.Errorf("expected size 256 found %d", rsaPubKey.Size())
 		}
 	})
+	t.Run("Time", func(t *testing.T) {
+		now := time.Now()
+		ts := strconv.Itoa(int(now.UnixNano()))
+		wallclock := ast.NumberTerm(json.Number(ts))
+
+		t.Run("if provided, is parsed properly", func(t *testing.T) {
+			c := ast.NewObject()
+			c.Insert(ast.StringTerm("time"), wallclock)
+			constraints, err := parseTokenConstraints(c, ast.NumberTerm(json.Number("12134")))
+			if err != nil {
+				t.Fatalf("parseTokenConstraints: %v", err)
+			}
+			diff := now.UnixNano() - constraints.time
+			if diff > 1000 {
+				t.Errorf("expected time constraint to be within 1000ns of the one passed in, got %dns", diff)
+			}
```

Comment:
Why do we need this subtraction? Can't we check if now.UnixNano() is equal to constraints.time? I would have expected the values to be exactly the same. Do we need to use `time.Now` at all?

Reply:
> I would have expected the values to be exactly the same

So would I. Anyways, there was a difference. Turns out we can just put some string "123" into it, and get it back here, and everything will be exact; the situation only differs, by some diff < 1ms, in the time.Now() case. I'll have a closer look once more!